src/redux/books/books.js

```diff
@@ -12,7 +30,9 @@ export default function booksReducer(state = [], action) {
       ];
 
     case REMOVE_BOOK:
-      return [...state.filter((book) => (book.id !== action.payload.id))];
+      return [...state.filter((book) => book.id !== action.payload.id)].map(
+        (book, id) => ({ ...book, id }),
+      );
```

Comment:
- [ ] You did a great job working on your remove book logic! 💯 However, you're making two operations (filter and map) that could be done just in one: 

```suggestion
      return state.filter((book) => book.id !== action.payload.id)
```

[`Array.filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) is a function that doesn't modify the original array, instead, it returns a new array with the elements that match the given condition, in this case, the elements that have a different id than the action payload, so it is Immutable. Please consider applying this change since it makes your code more readable and also more performant! 🤓 🛠️